### PR TITLE
Saga changes to publish registrations to downstream system.

### PR DESCRIPTION
**Pull request by sumathi-thirumani**
1. Changes to decouple create and publish from incoming message processing
2. Apply retry for publishing data to downstream system
3. Introduction of correlation key (assessment student ID) in saga table
4. Inclusion of status code entity
5. Saga cleanup based on #1
6. Unit testcases.

### CODE REVIEW

File: api/src/main/java/ca/bc/gov/educ/eas/api/service/v1/events/EventHandlerService.java [hunk 1]
```diff
@@ -38,118 +42,127 @@
...
-        .eventStatus(MESSAGE_PUBLISHED.toString())
-        .eventOutcome(event.getEventOutcome().toString())
-        .replyChannel(event.getReplyTo())
-        .build();
-  }
+    private EasEventEntity createEventRecord(Event event) {
+        return EasEventEntity.builder()
+                .createDate(LocalDateTime.now())
+                .updateDate(LocalDateTime.now())
+                .createUser(event.getEventType().toString()) //need to discuss what to put here.
```
Comment (arcshiftsolutions) [inline api/src/main/java/ca/bc/gov/educ/eas/api/service/v1/events/EventHandlerService.java]: EAS_API in the application properties can be used for update/create user.

File: api/src/test/java/ca/bc/gov/educ/eas/api/BaseEasAPITest.java [hunk 2]
```diff
@@ -55,9 +57,9 @@ public SessionEntity createMockSessionEntity() {
             .courseMonth(Integer.toString(currentDate.getMonthValue()))
             .activeFromDate(currentDate.minusMonths(2))
             .activeUntilDate(currentDate.plusMonths(2))
-            .createUser("ABC")
+            .createUser("EAS_API")
```
Comment (arcshiftsolutions) [inline api/src/test/java/ca/bc/gov/educ/eas/api/BaseEasAPITest.java]: Please grab this out of the application props

**Review by arcshiftsolutions — APPROVED** [review]

File: api/src/main/java/ca/bc/gov/educ/eas/api/service/v1/events/EventHandlerService.java (hunk 1 above)
Comment (sumathi-thirumani) [reply api/src/main/java/ca/bc/gov/educ/eas/api/service/v1/events/EventHandlerService.java]: Changes applied.

File: api/src/test/java/ca/bc/gov/educ/eas/api/BaseEasAPITest.java (hunk 2 above)
Comment (sumathi-thirumani) [reply api/src/test/java/ca/bc/gov/educ/eas/api/BaseEasAPITest.java]: Changes applied.

**Pull request merged**